pkg/storage/scrub.go

```diff
@@ -53,25 +56,38 @@ func (sc StoreController) CheckAllBlobsIntegrity() (ScrubResults, error) {
 	imageStoreList[""] = sc.DefaultStore
 
 	for _, imgStore := range imageStoreList {
-		images, err := imgStore.GetRepositories()
+		imgStoreResults, err := CheckImageStoreBlobsIntegrity(imgStore, nil)
 		if err != nil {
 			return results, err
 		}
 
-		for _, repo := range images {
-			imageResults, err := checkImage(repo, imgStore)
-			if err != nil {
-				return results, err
-			}
+		results.ScrubResults = append(results.ScrubResults, imgStoreResults...)
+	}
+
+	return results, nil
+}
 
-			results.ScrubResults = append(results.ScrubResults, imageResults...)
+func CheckImageStoreBlobsIntegrity(imgStore ImageStore, log *log.Logger) ([]ScrubImageResult, error) {
+	results := []ScrubImageResult{}
+
+	images, err := imgStore.GetRepositories()
+	if err != nil {
+		return results, err
+	}
+
+	for _, repo := range images {
+		imageResults, err := checkImage(repo, imgStore, log)
```

Comment:
So for each repo, we are lock/unlock and checking. This looks correct. Call this "checkRepo()" instead here.